services/storage/src/simcore_service_storage/access_layer.py

```diff
@@ -0,0 +1,266 @@
+""" Helper functions to determin access-rights on stored data
+
+# DRAFT Rationale:
+
+    osparc-simcore defines TWO authorization methods: i.e. a set of rules on what,
+    how and when any resource can be accessed or operated by a user
+
+    ## ROLE-BASED METHOD:
+        In this method, a user is assigned a role (user/tester/admin) upon registration. Each role is
+        system-wide and defines a set of operations that the user *can* perform
+            - Every operation is named as a resource and an action (e.g. )
+            - Resource is named hierarchically
+            - Roles can inherit permitted operations from other role
+        This method is static because is system-wide and it is defined directly in the
+        code at services/web/server/src/simcore_service_webserver/security_roles.py
+        It is defined on top of every API entrypoint and applied just after authentication of the user.
+
+    ## GROUP-BASED METHOD:
+        The second method is designed to dynamically share a resource among groups of users. A group
+        defines a set of rules that apply to a resource and users can be added to the group dynamically.
+        So far, there are two resources that define access rights (AR):
+            - one applies to projects (read/write/delete) and
+            - the other to services (execute/write)
+        The project access rights are set in the column "access_rights" of the "projects" table .
+        The service access rights has its own table: service_access_rights
+
+        Access rights apply hierarchically, meaning that the access granted to a project applies
+        to all nodes inside and stored data in nodes.
```

Comment:
this is only the case with this PR. before that the viewer could actually also save. thanks!